testing/fleetservertest/handlers.go

```diff
@@ -61,43 +58,180 @@ func NewPathUploadComplete(agentID string) string {
+		enrollRequest EnrollRequest) (*EnrollResponse, *HTTPError) {
 
-	policy := checkinResponseJSONPolicySystemIntegration
-	if withEndpoint {
-		policy = checkinResponseJSONPolicySystemIntegrationAndEndpoint
+		h.AgentID = agentID
+
+		localMetadata, err := json.Marshal(enrollRequest.Metadata.Local)
+		if err != nil {
+			return nil, &HTTPError{
+				StatusCode: http.StatusInternalServerError,
+				Message: fmt.Sprintf(
+					"could not marshal enroll request metadata into JSON: %v",
+					err),
+			}
+		}
+
+		localMetadata, err = updateLocalMetaAgentID(localMetadata, agentID)
+		if err != nil {
+			return nil, &HTTPError{
+				StatusCode: http.StatusInternalServerError,
+				Message:    fmt.Sprintf("could not update local metadata: %v", err),
+			}
+		}
+
+		return &EnrollResponse{
+			Action: "created",
+			Item: EnrollResponseItem{
+				AgentID:              agentID,
+				Active:               true,
+				PolicyID:             policyID,
+				Type:                 "PERMANENT",
+				EnrolledAt:           timeNow().Format(time.RFC3339),
+				UserProvidedMetadata: nil,
+				LocalMetadata:        localMetadata,
+				Actions:              nil,
+				AccessApiKeyID:       apiKey.ID,
+				AccessApiKey:         apiKey.Key,
+				Status:               "online",
+				Tags:                 enrollRequest.Metadata.Tags,
+			},
+		}, nil
 	}
+}
+
+// CheckinAction is the actions to be sent on next checkin and the delay, how
+// long the handler will wait before sending the response.
+type CheckinAction struct {
+	AckToken string
+	Actions  []string
+	Delay    time.Duration
+}
+
+// NewHandlerCheckinFakeComponent takes a generator function that returns the
+// actions to be sent on the next checkin. The actions format is a JSON list.
+// E.g.:
+//   - ["action1", "action2"]
+//   - ["action1"]
+//   - []
+func NewHandlerCheckinFakeComponent(next func() (CheckinAction, *HTTPError)) func(
+	ctx context.Context,
+	h *Handlers,
+	agentID string,
+	userAgent string,
+	acceptEncoding string,
+	checkinRequest CheckinRequest) (*CheckinResponse, *HTTPError) {
 
 	return func(
 		ctx context.Context,
-		id string,
+		h *Handlers,
+		agentID string,
 		userAgent string,
 		acceptEncoding string,
 		checkinRequest CheckinRequest) (*CheckinResponse, *HTTPError) {
+		if agentID != h.AgentID {
+			return nil, &HTTPError{
+				StatusCode: http.StatusNotFound,
+				Message:    fmt.Sprintf("agent %q not found", agentID),
+			}
+		}
 
+		data, hErr := next()
+		if hErr != nil {
+			return nil, hErr
+		}
+
+		respStr := NewCheckinResponse(data.AckToken, data.Actions...)
 		resp := CheckinResponse{}
 		err := json.Unmarshal(
-			[]byte(fmt.Sprintf(policy, ackToken, agentID)),
+			[]byte(respStr),
 			&resp)
 		if err != nil {
 			return nil, &HTTPError{
 				StatusCode: http.StatusInternalServerError,
-				Error:      err.Error(),
-				Message:    "failed to unmarshal policy",
+				Message:    fmt.Sprintf("failed to unmarshal policy: %v", err),
 			}
 		}
 
+		// simulate long pool
```

Comment:
I wish it was a pool :ocean: :beach_umbrella:  :wink: 
```suggestion
		// simulate long poll
```